lib/internal/modules/esm/loader.js

```diff
@@ -107,6 +108,13 @@ class Loader {
     }
 
     const { format } = getFormatResponse;
+    if (format === null) {
+      const dataUrl = RegExpPrototypeExec(
+        /^data:([^/]+\/[^;,]+)(?:[^,]*?)(;base64)?,/,
+        url,
+      );
+      throw new ERR_UNKNOWN_MODULE_FORMAT(dataUrl ? dataUrl[1] : 'null');
```

Comment:
```suggestion
      throw new ERR_INVALID_MODULE_SPECIFIER(dataUrl ? dataUrl[1] : 'null');
```

</br>

It seems like you are validating the [MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#types) based on the module specifier.

Perhaps [`ERR_INVALID_MODULE_SPECIFIER`](https://nodejs.org/api/errors.html#errors_err_invalid_module_specifier) would be more suitable in this case?

It is entirely possible that we would benefit from a more descriptive error code for invalid MIME types&hellip;